bemuse/src/app/ui/OptionsAdvanced.tsx

```diff
@@ -20,6 +20,9 @@ const parseLatency = (latencyText: string): number => {
 const useLatency = (handler: (latency: number) => void) => {
   useEffect(() => {
     const handleMessage = (event: MessageEvent) => {
+      if (event.origin !== 'https://bemuse.ninja') {
+        throw new Error('invalid origin')
+      }
```

Comment:
I think we can allow message from any origin, because with this hardcoding we cannot test on localhost without changing code.

If we really want to make it secure, one option is to generate a UUID ([crypto.randomUUID()](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID)) and pass it when opening the window. Then onmessage can check if the the correct UUID is being passed back to Bemuse.

Reply:
ok, reverted on cf14dd98. This change came from the vulnerability report by Sonarclould.

Reply:
I see... I will annotate this.

<img width="844" alt="image" src="https://user-images.githubusercontent.com/193136/210934409-2c50a70b-d028-4794-97ae-1b443b2fc4be.png">